core-api/src/entities/permission.entity.ts

```diff
@@ -0,0 +1,92 @@
+import {
+    Column,
+    Entity,
+    PrimaryGeneratedColumn,
+    CreateDateColumn,
+    UpdateDateColumn,
+} from "typeorm";
+import { Length } from "class-validator";
+import Messages from "../utils/messages";
+
+/**
+ * @namespace Entities
+ * @class User
+ */
+@Entity()
+export class Permisson {
+    @PrimaryGeneratedColumn()
+    private id: number;
+
+    @Column({ length: 64 })
+    @Length(3, 20, {
+        message: Messages.validation.permisson_length,
+    })
+    private name: string;
+
+    @Column()
+    private softDeleted: boolean;
+
+    @CreateDateColumn()
+    private createdAt: Date;
+
+    @UpdateDateColumn()
+    private updatedAt: Date;
+
+    constructor(name: string, softDeleted = false) {
+        this.name = name;
+        this.softDeleted = softDeleted;
+    }
+
+    public getId(): number {
+        return this.id;
+    }
+
+    public getName(): string {
+        return this.name;
+    }
+
+    public setName(name: string): string {
+        this.name = name;
+        return this.name;
+    }
+    public isSoftDeleted(): boolean {
+        return this.softDeleted;
+    }
+
+    public setSoftDeleted(softDeleted: boolean): boolean {
+        this.softDeleted = softDeleted;
+        return this.softDeleted;
+    }
+
+    public getCreatedAt(): Date {
+        return this.createdAt;
+    }
+
+    public getUpdatedAt(): Date {
+        return this.updatedAt;
+    }
+
+    public toSafe(): SafePermisson {
+        return new SafePermisson(this);
+    }
+
+}
+
+/**
+ * @namespace Entities
+ * @class SafePermisson
+ */
+export class SafePermisson {
```

Comment:
I think this is not necessary... `SafeUser` was a workaround for hiding user password and other data, here you aren't doing anything special with `SafePermission`. I think it's totally safe (haha) to return plain `Permission` objects in json responses.

Reply:
Permission has the soft Deleted attribute that is not good to be return, so this way is a workaround for hiding it.